### PR TITLE
Feat/region-wide-alert

### PR DESCRIPTION
**Pull request by Ahmedhossamdev**
fixes: #78

### CODE REVIEW

File: src/routes/api/oba/alerts/+server.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		);
+	}
+}
+
+function isValidAlert(alert) {
+	return isAgencyWideAlert(alert) && isStartDateWithin24Hours(alert) && isHighSeverity(alert);
+}
+
+function isHighSeverity(alert) {
+	const isHighSeverity =
```
Comment (aaronbrethorst) [inline src/routes/api/oba/alerts/+server.js:45]: let's refactor this with early returns:

```
function isHighSeverity(alert) {
	if (!alert) {
		return false;
	}

	const severity = getSeverityLevel(alert);
	return severity === GtfsRealtimeBindings.transit_realtime.Alert.SeverityLevel.SEVERE ||
		  severity === GtfsRealtimeBindings.transit_realtime.Alert.SeverityLevel.WARNING;
}
```

File: src/routes/api/oba/alerts/+server.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+function isStartDateWithin24Hours(alert) {
+	const startDate = alert.activePeriod[0].start;
+	const now = Date.now() / 1000;
+	return startDate <= now && startDate >= now - 24 * 60 * 60;
+}
+
+function isAgencyWideAlert(alert) {
+	return alert.informedEntity && alert.informedEntity.length > 0;
```
Comment (aaronbrethorst) [inline src/routes/api/oba/alerts/+server.js:65]: is this only `true` under the same circumstances as on Android? https://github.com/OneBusAway/onebusaway-android/pull/1274/files#diff-aeef62db59a9689cfc54bcdff19bfd5934dd4846e06c6a47f6d497bb7d25244fR108

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
great work! I have a couple questions and small tweaks I'd like to see but otherwise I think this is ready to go :)

File: src/routes/api/oba/alerts/+server.js (hunk 2 above)
Comment (Ahmedhossamdev) [reply src/routes/api/oba/alerts/+server.js:65]: Yes it's the same as 

`
GtfsRealtimeBindings.transit_realtime.EntitySelector.fromObject(alert.informedEntity[0])
`

We should return true only if the agency ID exists, right?

**Pull request merged**